lib/matplotlib/widgets.py

```diff
@@ -182,13 +182,29 @@ def __init__(self, ax, label, image=None,
         self.connect_event('button_press_event', self._click)
         self.connect_event('button_release_event', self._release)
         self.connect_event('motion_notify_event', self._motion)
+        self.connect_event('draw_event', self._draw)
+        self.connect_event('resize_event', self._clear)
         ax.set_navigate(False)
         ax.set_facecolor(color)
         ax.set_xticks([])
         ax.set_yticks([])
         self.color = color
         self.hovercolor = hovercolor
 
+        self.useblit = useblit and self.canvas.supports_blit
+        if self.useblit:
+            self.label.set_animated(True)
+            self.ax.patch.set_animated(True)
+            self.background = self.canvas.copy_from_bbox(self.ax.bbox)
```

Comment:
I do not think it is safe to assume that we have drawn the figure at least once here, we need to defer this to the first draw event.